syntax = "proto3";

package calculator;

// Principal
service CentralService {
  rpc ProcessNumbers (ClientRequest) returns (ClientResponse);
}

// Suma
service SumService {
  rpc SumarNumeros (DosNumeros) returns (Resultado);
}

// Multi
service MultiplyService {
  rpc MultiplicarNumeros (DosNumeros) returns (Resultado);
}

message ClientRequest {
  int32 num1 = 1;
  int32 num2 = 2;
  int32 num3 = 3;
}

message ClientResponse {
  int32 result = 1;
}

message DosNumeros {
  int32 num1 = 1;
  int32 num2 = 2;
}

message Resultado {
  int32 result = 1;
}